plugins/woocommerce-admin/client/inbox-panel/test/index.js

```diff
@@ -114,10 +149,67 @@ describe( 'inbox_panel_view_event', () => {
 			notesHaveResolved: true,
 			isBatchUpdating: false,
 		} ) );
+		InboxNoteCard.mockImplementation( ( { note } ) => {
+			return <div>{ note.id }</div>;
+		} );
 		render( <InboxPanel /> );
 
 		expect( recordEvent ).toHaveBeenCalledWith( 'inbox_panel_view', {
 			total: 5,
 		} );
 	} );
 } );
+
+describe( 'inbox_note_view event', () => {
+	test( 'should fire when inbox note card calls onNoteVisible', () => {
+		notes.forEach( ( note ) => ( note.is_deleted = false ) );
+		useSelect.mockImplementation( () => ( {
+			notes,
+			isError: false,
+			notesHaveResolved: true,
+			isBatchUpdating: false,
+		} ) );
+		InboxNoteCard.mockImplementation( ( { onNoteVisible, note } ) => {
+			useEffect( () => onNoteVisible( note ), [] );
+			return <div>{ note.id }</div>;
+		} );
```

Comment:
Why did this have be be mocked?

Reply:
It has some sort of visibility detector that wasn't working on the tests, so I thought it was best to just mock it to call onNoteVisible.

Reply:
How about adding a comment in the code to explain why this is being done?